tests/sort-demo/TestBoard.pas

```diff
@@ -41,11 +44,47 @@   TestTBoard = class(TTestCase)
 
 implementation
 
+procedure TestTBoard.UstawDaneWBoard(AData: TArray<Integer>);
+var
+  idx: Integer;
+begin
+  FBoard.GenerateData(Length(AData));
+  for idx := 0 to Length(AData) - 1 do
+    FBoard.Data[idx] := AData[idx];
+end;
+
+procedure TestTBoard.WykonajSortowanie(AProc: TProc);
+var
+  sortThread: TThread;
+  done: Boolean;
+begin
+  done := False;
+  sortThread := TThread.CreateAnonymousThread(
+    procedure
+    begin
+      AProc;
+      done := True;
+    end);
+  sortThread.FreeOnTerminate := True;
+  sortThread.Start;
+  while not done do
+    sleep(10);
+end;
+
 procedure TestTBoard.SetUp;
 begin
   FBoard := TBoard.Create;
 end;
 
+procedure TestTBoard.SprawdzDaneWBoard(AExpectedData: TArray<Integer>);
+var
+  idx: Integer;
+begin
+  for idx := 0 to Length(AExpectedData) - 1 do
+    CheckEquals(AExpectedData[idx], FBoard.Data[idx],
+      'Nieprawidłowa wartość danych w tabeli na idx = ' + idx.ToString);
+end;
```

Comment:
Czy funkcja nie powinna na początku sprawdzać czy te tablice mają taki sam rozmiar?

Reply:
świetna uwaga 👍

Przy okazji warto się wyćwiczyć w krótkim komentowaniu, bo pisanie esejów w CodeReview to słaby pomysł, podobnie trzeba unikać flame-war-a: https://en.wikipedia.org/wiki/Flaming_(Internet)

tutaj można napisać:
`może: porównać rozmiary tablic?`
albo tak
`a weryfikacja rozmiarów obu tablic`

Czytający przegląd, czyli autor kodu, powinien wiedzieć co z tym zrobić
oczywiście na początku warto zaczynać komentarz od: 
* `warto dodać ...`,
* `ja bym zrobił tak ...`, 
* `spróbuj w ten sposób ...`, 
* `może ...`, 
* `IMHO ...`, 
aby nie zostać odebranym jako atakujący. Jednak z czasem te zwroty grzecznościowe tracą wartość